### PR TITLE
Crm 16259 1

### PR DESCRIPTION
**Pull request by pradpnayak**

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -2999,6 +2999,11 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         $balanceTrxnParams['status_id'] = $statusId;
         $balanceTrxnParams['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
         $balanceTrxnParams['check_number'] = CRM_Utils_Array::value('check_number', $params);
+        if (!empty($balanceTrxnParams['from_financial_account_id']) &&
+          ($statusId == array_search('Completed', $contributionStatuses) || $statusId == array_search('Partially paid', $contributionStatuses))
+        ) {
+          $balanceTrxnParams['is_payment'] = 1;
```
Comment (JoeMurray) [inline CRM/Contribute/BAO/Contribution.php]: May be an issue if this is due to an edit of contribution. Please check

**Comment by JoeMurray** [comment]
It says 1 failing check but Details shows everything is fine. Going to retest.

**Comment by JoeMurray** [comment]
jenkins, test this please

**Comment by colemanw** [comment]
@mlutfy it looks like there's a server error causing tests to crash.

**Comment by mlutfy** [comment]
@colemanw where? I see a test job currently running (https://test.civicrm.org/job/CiviCRM-Core-Matrix/CIVIVER=4.5,label=ubuntu1204/1921/console)

**Comment by colemanw** [comment]
@mlutfy https://test.civicrm.org/job/CiviCRM-Core-PR/7138/console

**Comment by davecivicrm** [comment]
Jenkins, retest this please.

**Comment by davecivicrm** [comment]
This looks good. Tested full 'payment' against contribution, membership, event as well as event partial payment scenario.

**Pull request merged**